staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -435,6 +436,14 @@ func summarizeData(data []byte, maxLength int) string {
 
 func limitedReadBody(req *http.Request, limit int64) ([]byte, error) {
 	defer req.Body.Close()
+
+	// Get Content-Length from header to avoid unnecessary I/O operation
+	if contentLength, err := strconv.Atoi(req.Header.Get("Content-Length")); err == nil {
+		if int64(contentLength) > limit {
+			return nil, errors.NewRequestEntityTooLargeError(fmt.Sprintf("limit is %d", limit))
+		}
+	}
+
```

Comment:
I don't think the changes is right. As if Content-Length is the length ot whole reponse message. In some case, we should limit the length of reading by limit.

Reply:
Ref： https://golang.org/pkg/io/#LimitedReader ”A LimitedReader reads from R but limits the amount of data returned to just N bytes.“。 Many functions calling limitedReadBody with pass limit. Are you sure that the value of limit is always equivalent to Content-Length?

Reply:
And if we want to read all content of the body, we can only call ioutil.ReadAll(req.Body).